src/map/latent_effect_container.cpp

```diff
@@ -717,25 +717,50 @@ bool CLatentEffectContainer::ProcessLatentEffect(CLatentEffect& latentEffect)
             break;
         case LATENT::SIGNET_BONUS:
         {
+            // player is logging in/zoning
```

Comment:
Question about these: 
Do we _ever_ want to be trying to process a latent on a `PChar` if their zone is `nullptr`? 
Should we be putting a single check at the top of this function call?
```cpp
if (m_POwner->loc.zone == nullptr)
{
    break;
}
```

Reply:
lets just check it before the switch case so we don't need to repeat it in each case, because if you haven't got a zone shit has already hit the fan anyway